src/datatrove/pipeline/readers/base.py

```diff
@@ -185,10 +190,19 @@ def read_files_shard(self, shard: list[str]) -> DocumentsPipeline:
         """
         li = 0
         skipped = 0
-        with tqdm(total=self.limit if self.limit != -1 else None) if self.progress else nullcontext() as pbar:
-            for filepath in shard:
+        with (
+            tqdm(
+                total=self.limit if self.limit != -1 else None,
+                desc="Document progress",
+                unit="doc",
+                disable=not self.doc_progress,
+            ) as doc_pbar,
+            tqdm(total=len(shard), desc="File progress", unit="file", disable=not self.file_progress) as file_pbar,
+        ):
+            for i, filepath in enumerate(shard):
                 self.stat_update("input_files")
-                logger.info(f"Reading input file {filepath}")
+                logger.info(f"Reading input file {filepath}, {i+1}/{len(shard)}")
+                file_pbar.update()
```

Comment:
I think this should be after the loop with all the docs, so that we only show one file processed after we finish the entire file

Reply:
Not sure if I understand. You think it would be better to update the progress bar after we actually processed the file ?

Reply:
exactly